tests/rados/test_reads_balancer.py

```diff
@@ -0,0 +1,273 @@
+                )
+                try:
+                    out, err = client_node.exec_command(sudo=True, cmd=change_cmd)
+                except Exception as e:
+                    if error_str not in str(e).strip():
+                        log.error(
+                            "Correct error string not seen in stderr stream"
+                            f"Expected : {error_str} \n Got : {e}"
+                        )
+                        raise Exception("Wrong error string obtained")
+                    log.error(f"An error occurred but was expected: {e}")
+                    log.debug(
+                        "Could not change the PG set with OSD which is not present in the acting set.. Pass"
+                    )
+
+                # Trying to change the primary with secondary. Should be possible
+                pgid = f"{pool_id}.0"
+                acting_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+                new_osd = acting_set[1]
+                change_cmd = f"ceph osd pg-upmap-primary {pgid} {new_osd}"
+                error_str = f"change primary for pg {pgid} to osd.{new_osd}"
+                out, err = client_node.exec_command(sudo=True, cmd=change_cmd)
+                if error_str not in err.strip():
+                    log.error(
+                        "Correct error string not seen in stderr stream"
+                        f"Expected : {error_str} \n Got : {err}"
+                    )
+                    raise Exception("wrong error string obtained")
+
+                log.debug("Successfully changed the PG set with OSD.. Pass")
+
+                # Checking the change in the new acting set
+                new_acting_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+                if new_acting_set[0] != acting_set[1]:
+                    log.error(
+                        f"The acting set has not changed."
+                        f"Old AC : {acting_set}, New AC : {new_acting_set}"
+                    )
+                    raise Exception("AC not changed error")
+
+                time.sleep(5)
+                # Checking if it is possible to revert the changes made previously via "pg-upmap-primary"
+                rm_cmd = f"ceph osd rm-pg-upmap-primary {pgid}"
+                out, err = client_node.exec_command(sudo=True, cmd=rm_cmd)
+                # Checking the change in the new acting set
+                time.sleep(20)
+                final_acting_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+                if final_acting_set[0] != acting_set[0]:
+                    log.error(
+                        f"The acting set has not reverted to original."
+                        f"Old AC : {acting_set}, final AC : {final_acting_set}"
+                    )
+                    raise Exception("AC not changed error")
+
+                log.info("Completed tests on replicated pools for online commands")
+                break
```

Comment:
If we break here, we would be checking this workflow for only one replicated pool, I hope that's the intention.
Also if that's the intention, maybe instead of going through a _for_ loop, we can choose the first replicated pool we created and run the whole _online_command_verification_ workflow for that pool itself

Reply:
Just kept it like that so that in future, after completing the entire workflow of the reads balancer, i can run it on all the pools.

I can remove if needed, but i'm inclined to keep it.